src/libraries/Microsoft.Extensions.Logging.Console/src/FormatterOptionsMonitor.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.Logging.Console
+{
+    internal class FormatterOptionsMonitor<TOptions> : IOptionsMonitor<TOptions> where TOptions : ConsoleFormatterOptions
```

Comment:
Is this for tests?

Reply:
in ConsoleLoggerProvider... in case DI picks

        public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options)
            : this(options, Enumerable.Empty<ConsoleFormatter>()) { }

then SetFormatters would get called using FormatterOptionsMonitor... I think perhaps this should be simplified and not use FormatterOptionsMonitor?

Reply:
DI will pick the longest constructor.

Reply:
UPDATE:
The aim here is to make sure if ConsoleLoggerProvider is manually constructed using the shorter ctor, then we make sure to add the built-in formatters too.

- [ ] Make sure formatters get change notification in the manually constructed workflow.

Reply:
@maryamariyan it doesn't need to figure for manually constructed. The caller should figure that out on their own.